src/plots/barPlot.ts

```diff
@@ -439,10 +439,13 @@ export module Plots {
 
       let scale = <QuantitativeScale<any>>accScaleBinding.scale;
 
-      extents = extents.map((extent) => [
+      // To account for inverted domains
+      extents = extents.map((extent) => d3.extent([
         scale.invert(scale.scale(extent[0]) - this._barPixelWidth / 2),
-        scale.invert(scale.scale(extent[1]) + this._barPixelWidth / 2),
-      ]);
+        scale.invert(scale.scale(extent[0]) + this._barPixelWidth / 2),
+        scale.invert(scale.scale(extent[1]) - this._barPixelWidth / 2),
+        scale.invert(scale.scale(extent[1]) + this._barPixelWidth / 2)
```

Comment:
Checking to see if I understand.
1. Extent is what you would get by autodomaining on any other type of plot.
2. It used to worked/works on vertical plots because extent[0] is always <= extent[1] (assumption), so you could subtract 1/2 * barPixelWidth from extent[0], and add it to extent[1], and get a strictly wider range
4. So on horizontal, can extent[1] be less than extent[0]? So you account for that by always subtracting and adding from both extent[0] and extent[1]?